### DISCUSSION TITLE
New uvicorn[standard] version cause memory leak due to tcp ping

### DISCUSSION THREAD

**Discussion by KiraPC** [tiangolo/uvicorn-gunicorn-docker#231] [Questions] [answered]
As I described in the uvicorn issue https://github.com/encode/uvicorn/issues/1226 using standard uvicorn version from 0.14.0 tcp ping cause memory leak that end with OOM.

Since this image use that version, it should be affected by the leak.

Someone faced this problem?

**Comment by edsonlalmeida** [comment]
i have same a problem in my app

**Comment by nvtkaszpir** [comment]
I guess we see something similar with kubernetes standard tcp probe where it just checks if port is open.

**Comment by tiangolo** [comment, marked as answer]
The best place for the conversation about it is probably there in the Uvicorn repo. I see that issue was closed, so I suspect it's solved now. :tada: 

Also, just in case, now that Uvicorn supports managing workers with `--workers`, including restarting dead ones, there's no need for Gunicorn. That also means that it's much simpler to build a Docker image from scratch now, I updated the docs to explain it.

Because of that, I deprecated this Docker image: https://github.com/tiangolo/uvicorn-gunicorn-docker#-warning-you-probably-dont-need-this-docker-image
